var/spack/repos/builtin/packages/py-pymol/package.py

```diff
@@ -15,29 +15,42 @@ class PyPymol(PythonPackage):
     homepage = "https://pymol.org"
     url      = "https://github.com/schrodinger/pymol-open-source/archive/v2.4.0.tar.gz"
 
+    version('2.5.0', sha256='aa828bf5719bd9a14510118a93182a6e0cadc03a574ba1e327e1e9780a0e80b3')
     version('2.4.0', sha256='5ede4ce2e8f53713c5ee64f5905b2d29bf01e4391da7e536ce8909d6b9116581')
     version('2.3.0', sha256='62aa21fafd1db805c876f89466e47513809f8198395e1f00a5f5cc40d6f40ed0')
 
-    depends_on('python+tkinter', type=('build', 'run'))
-    depends_on('freetype', type=('build', 'run'))
-    depends_on('glew', type=('build'))
-    depends_on('glm', type=('build'))
-    depends_on('py-numpy', type=('build', 'run'))
-    depends_on('tcsh', type=('build', 'run'))
+    depends_on('python+tkinter@2.7:', type=('build', 'run'), when='@2.3.0:2.4.0')
+    depends_on('python@3.6:', type=('build', 'run'), when='@2.5.0:')
+    depends_on('gl')
+    depends_on('glew')
+    depends_on('libpng')
+    depends_on('freetype')
+    depends_on('glm')
+    depends_on('libmmtf-cpp')
+    depends_on('msgpack-c@2.1.5:')
+    depends_on('netcdf-cxx4')
+    depends_on('libxml2')
     depends_on('py-pyqt5', type=('build', 'run'))
-    depends_on('py-pmw', type=('build', 'run'))
-    depends_on('libmmtf-cpp', type=('build', 'run', 'link'))
-    depends_on('msgpack-c', type=('build', 'run'))
-    depends_on('libpng@1.5.13', type=('build', 'run'))
-
-    def setup_build_environment(self, env):
-        include = []
-        library = []
-        for dep in self.spec.dependencies(deptype='link'):
-            query = self.spec[dep.name]
-            include.extend(query.headers.directories)
-
-        env.set('CPATH', ':'.join(include))
-        env.set('LIBRARY_PATH', ':'.join(library))
-        env.set('PREFIX_PATH', self.spec['libpng'].prefix)
-        env.prepend_path('PREFIX_PATH', self.spec['py-pyqt5'].prefix)
+    depends_on('py-numpy', type=('build', 'run'))
+
+    def install_args(self, spec, prefix):
+        args = super(PyPymol, self).install_args(spec, prefix)
+        args.append('--no-launcher')
```

Comment:
Remind me to update this in #27798 after this is merged

Reply:
will do